lib/src/screens/setup_2fa/modify_2fa_page.dart

```diff
@@ -39,17 +77,124 @@ class Modify2FAPage extends BasePage {
                         Navigator.of(context).pop();
                       },
                       actionRightButton: () {
-                        setup2FAViewModel.setUseTOTP2FA(false);
+                      widget.setup2FAViewModel.setUseTOTP2FA(false);
                         Navigator.pushNamedAndRemoveUntil(
                             context, Routes.dashboard, (route) => false);
                       },
                     );
                   },
                 );
-              }),
+            },
+          ),
+          StandardListSeparator(padding: EdgeInsets.symmetric(horizontal: 24)),
+          SizedBox(height: 40),
+          Padding(
+            padding: EdgeInsets.symmetric(horizontal: 24),
+            child: Text(
+              'Cake 2FA Preset',
+              style: TextStyle(
+                fontSize: 14,
+                fontWeight: FontWeight.normal,
+                color: Theme.of(context).primaryTextTheme.titleLarge!.color!,
+              ),
+            ),
+          ),
+          SizedBox(height: 8),
+          Container(
+            padding: EdgeInsets.symmetric(horizontal: 6, vertical: 4),
+            margin: EdgeInsets.symmetric(horizontal: 24),
+            decoration: BoxDecoration(
+              color: Color(0xffF2F0FA),
+              borderRadius: BorderRadius.circular(20.0),
+            ),
+            child: Theme(
+              data: ThemeData(
+                  primaryTextTheme: TextTheme(
+                      bodyLarge:
+                          TextStyle(backgroundColor: Colors.transparent))),
+              child: TabBar(
+                onTap: (value) => viewModel.selectCakePreset(value),
+                controller: _tabController,
+                indicator: BoxDecoration(
+                  borderRadius: BorderRadius.circular(25.0),
+                  color: !viewModel.unhighlightTabs
+                      ? Theme.of(context).accentTextTheme.bodyLarge!.color!
+                      : Colors.transparent, 
+                ),
+                labelColor: Theme.of(context)
+                    .primaryTextTheme
+                    .displayLarge!
+                    .backgroundColor!,
+                unselectedLabelColor: Theme.of(context)
+                    .primaryTextTheme
+                    .displayLarge!
+                    .backgroundColor!,
+                tabs: [
+                  Tab(text: S.current.narrow, height: 30),
+                  Tab(text: S.current.normal, height: 30),
```

Comment:
is there a specific reason why not directly using the already existing one? `SettingsChoicesCell`

Reply:
Didn't know of the `SettingsChoicesCell` previously.

Switched this widget to it and adjusted the code.